docs/specs/user-profile/design.md

```diff
@@ -237,6 +191,86 @@ The UI control of the custom attribute is determined by the `type` and the `form
 - `type: string` and `format: uri` is `<input type="text">` with validation.
 - `type: string` and `format: date-time` is rendered using a library.
 
+## Access control
+
+Access control on standard attributes and custom attributes are defined per party.
+There are 4 parties.
+
+### The end-user
+
+The end-user can view or edit the standard attributes and the custom attributes via the settings page.
+
+### The session bearer
+
+The session bearer is someone who has a valid IDP session cookie or a valid access token.
+The session bearer can then access the User Info endpoint and the resolver endpoint.
+
+Behind the session bearer, it is the end-user, the mobile app or the website.
+
+### The admin user
+
+The admin user can view or edit the standard attributes and the custom attributes via the portal.
+Behind the scene, the portal uses the Admin API.
+
+### The Admin API
+
+The Admin API allows the developer to view or edit the standard attributes and the custom attributes.
+The admin API always have full access to all standard attributes and the custom attributes.
+
+### Available access control levels
+
+The access control levels are defined as follows.
+They are ordered by in increasing order.
+
+- `hidden`: Hidden from the party.
+- `readonly`: Read-only to the party.
+- `readwrite`: Read-write to the party.
+
+### Access control configuration
+
+- The default access control level for `admin_user` is `readwrite`.
+- The default access control level is `bearer` is `readwrite`.
+- The access control level `readwrite` is equivalent to `readonly` for `bearer`.
+- The default access control level of standard attribute for `end_user` is `readwrite`.
+- The default access control level of custom attribute for `end_user` is `hidden`.
+- The access control level for `end_user` must be equal or less than that that for `bearer`.
+- The access control level for `bearer` must be equal or less than that for `admin_user`.
+
+Here is an example with default values shown.
+
+```yaml
+user_profile:
+  standard_attributes:
+    access_control:
+    - pointer: /given_name
+      access_control:
+        end_user: readwrite
+        bearer: readwrite
+        admin_user: readwrite
+  custom_attributes:
+    access_control:
+    - pointer: /hobby
+      access_control:
+        end_user: hidden
+        bearer: readwrite
+        admin_user: readwrite
+```
+
+### Exhaustive list of access control combination
+
+|end\_user|bearer|admin\_user|
+|---------|------|-----------|
+|hidden|hidden|hidden|
+|hidden|hidden|readonly|
+|hidden|hidden|readwrite|
+|hidden|readonly|readonly|
+|hidden|readonly|readwrite|
+|hidden|readwrite|readwrite|
+|readonly|readonly|readonly|
+|readonly|readonly|readwrite|
+|readonly|readwrite|readwrite|
+|readwrite|readwrite|readwrite|
```

Comment:
`readwrite` in the bearer means we will have api in the SDK to allow update user profile?
If yes, would be great if we can defined the SDK api in the spec.

Reply:
No. `readwrite` is equivalent to `readonly` in this case. See https://github.com/authgear/authgear-server/pull/1498/files#diff-299c7f857be8e61cccc8b09010cc5a0399272250bfeb8a4aeaff53292500b863R257

Reply:
Thanks for clarifying, LGTM:)